scripts/evaluate_model.py

```diff
@@ -0,0 +1,126 @@
+import argparse
+
+from datasets import load_dataset
+
+import numpy as np
+
+import torch
+from torch.utils.data import Dataset, DataLoader
+from torchtext.data.metrics import bleu_score
+
+from tqdm import tqdm
+
+from transformers import AutoTokenizer, AutoModelForSeq2SeqLM
+
+
+class ConvAI2CandidatesDataset(Dataset):
+    """ConvAI2 Candidates Dataset"""
+
+    def __init__(self, dataset):
+        """
+        Args:
+            dataset (DatasetDict): ConvAI2 dataset from huggingface
+        """
+
+        np.random.seed(42)
+        candidates_dataset = {"context": [], "candidates": []}
+        ids = list(range(len(dataset["train"])))
+
+        for i, dialog in (pbar := tqdm(enumerate(dataset["train"]))):
+            pbar.set_description("Creating dataset")
+            context = ""
+
+            for utterance in dialog["dialog"]:
+                if utterance["sender_class"] == "Bot" and context:
+                    candidates_dataset["context"].append(context)
+                    candidates = [utterance["text"]]
+                    dialog_ids = [i]
+
+                    while i in dialog_ids:
+                        dialog_ids = np.random.choice(ids, size=19, replace=False)
+
+                    for id in dialog_ids:
+                        other_dialog = dataset["train"][int(id)]["dialog"]
+                        length = len(other_dialog)
+                        text = ""
+                        sender = "Human"
+
+                        while sender == "Human":
+                            j = np.random.randint(length)
+                            text = other_dialog[j]["text"]
+                            sender = other_dialog[j]["sender"]
+                        candidates.append(text)
+
```

Comment:
Have some questions here:
- Negative candidates never will be from the same dialog. Is it okay?
- Negative candidates always bot messages. Is it also okay?

I also suggest you to simplify this logic: let's create an array with all possible negative candidates and the just sample from it.